### ISSUE TITLE
[NewsPublish] Question

### ISSUE THREAD

**Issue by Menteroth** [SharkyTheKing/Sharky#69]
Would it be possible to differentiate self, other bots and users?
So you can for instance have the bot only publish its own messages, self and other bots or just other users.

**Comment by SharkyTheKing** [comment]
Christ, I only just saw this. My apologizes.

I'm a bit confused by what you mean. Are you asking to make it so it can only publish itself, publish itself and other bots, and just publish everything except bots?

**Comment by Menteroth** [comment]
I almost forgot what I meant haha. I think it was to maybe have an option to toggle the three different things so it doesn't just publish everything posted unless all the options are set to `on`.

Publish users: on/off
Publish other bots: on/off
Publish self: on/off

I hope this makes it more clear.

**Comment by SharkyTheKing** [comment]
Oooh. I like this. When I find time I'll work on getting this done.

Or if you or someone else who happens to see this doesn't mind making a PR I'll happily look at that for this.

Wonderful suggestion! :)

**Comment by Menteroth** [comment]
Thanks. No rush at all. Glad I could help improve with the suggestion.

Not sure if this would be of use but if you wanted and it's possible you could maybe make options to set specific users and/or roles as well. If someone doesn't want the whole group of posters to get published posts. If that makes sense.

**Comment by SharkyTheKing** [comment]
True. Though the way I figure is, the news channels are usually locked down by the owner/admins of the server. They'll usually allow people to post in if they're trusted. So the specific users/roles I may not do, as you can specifically deny the people from the channel if needed.

**Comment by Menteroth** [comment]
Yeah that's a good point.
